pkg/kubectl/apply/element.go

```diff
@@ -36,6 +36,9 @@ type Element interface {
 	// Returns the Result of the merged elements
 	Merge(Strategy) (Result, error)
 
+	// Compare detects conflicts between recorded and remote values.
+	Compare(Strategy) bool
```

Comment:
i am not familiar with the code, but this doesn't seem good. Compare should have it's own strategy right?